### DISCUSSION TITLE
GraphQL Query finding sites multiple tags (OR)

### DISCUSSION THREAD

**Discussion by arjenvri** [netbox-community/netbox#12004] [Q&A] [answered]
I am trying to query sites using the netbox GraphQL interface by matching on tags.
Finding any site that has one specific tag works, but i would like to find sites that have tag x or sites that have tag y.

works
`
query {
  site_list(tag: "mytag1") {
    id
    name
  }
}
`

does not give any results (would expect multiple sites some with mytag1 and others with mytag2
`
query {
  site_list(tag: ["mytag1", "mytag2"]) {
    id
    name
  }
}
`

Similar query does seem to work for other fields. Any idea how i can accomplish this?

**Comment by candlerb** [comment, marked as answer]
> does not give any results (would expect multiple sites some with mytag1 and others with mytag2

I would expect it to return only those sites which have *both* mytag1 *and* mytag2.  This is how queries via the REST API work at least, as documented here:

https://docs.netbox.dev/en/stable/reference/filtering/#filtering-objects

<i>a logical AND operation will be used in instances where a field may have multiple values, such as tags. For example, `GET /api/dcim/sites/?tag=foo&tag=bar` will return only sites which have both the "foo" and "bar" tags applied.</i>

(and I believe the filtering logic is common between REST and graphql).

See also discussion #11703

**Reply by candlerb** [reply]
Confirmed by testing (v3.4.6)

```
# curl -H "Authorization: Token $TOKEN" -H "Content-Type: application/json" -H "Accept: application/json" "$NETBOX/graphql/" \
  --data '{"query": "{ site_list(tag: [\"mytag1\",\"mytag2\"]) { id name } }"}'
```

```
--- site 'Test' has tag 'mytag1' only ---
{"data":{"site_list":[]}}

--- site 'Test' has tags 'mytag1' and 'mytag2' ---
{"data":{"site_list":[{"id":"13","name":"Test"}]}}
```

**Reply by arjenvri** [reply]
Thank you @candlerb ! I confirmed indeed it works as AND now as well. I am still hoping for a query that does OR… any idea?

**Reply by candlerb** [reply]
https://github.com/netbox-community/netbox/discussions/11703#discussioncomment-4920796

You can merge the results client-side.  Unless there is a GraphQL equivalent to a SQL "union" to concatenate results server-side? (Searching for this gives union types, which is not what I was looking for)
